src/usermgmt.rs

```diff
@@ -78,14 +78,19 @@ fn create_session_cookie<'a>(
 pub struct CreateUserQ {
     email: String,
     password: String,
+    beta_key: String,
 }
 
 pub async fn create_user(
     q: CreateUserQ,
     pool: DB,
     pepper: impl Deref<Target = impl AsRef<[u8]>>,
     domain: impl Deref<Target = impl Clone + Into<Cow<'_, str>>>,
+    beta_key: impl Deref<Target = impl Clone + Into<Cow<'_, str>>>,
```

Comment:
The `Cow` was required for `domain` because of how it's used by `cookie`, it feels like for the beta key check the parameter type could be simpler.

Reply:
You have more experience than I do balancing generality in rust -- what did you have in mind? `impl Deref<Target = String>`? 

It seems like we could probably `Box::leak` it and get rid of the `Arc` down to just `&str`, but I'm not sure unidiomatic that is :)

Reply:
`impl Deref<Target = String>` is quite fine. As for `Box::leak`, I've no idea how idiomatic it is (probably not at all, given Rust's focus on memory safety), but it's very tempting — it'd allow us to get rid of Deref and Clone as well. We can try it if you want :)

Reply:
I don't know what you mean, leaking memory is perfectly safe :p Anyways, it's essentially leaked already, so I see we try it out! Completely gets rid of `Arc` and `Deref` as you say. 

I think the main risk is that `tokio::main` or `warp::serve` don't work like I expect, but using it locally I don't see an issue and the testsuite still passes (though it doesn't really exercise the multi-threading portion). Calling it a ton as fast as I can through curl nothing explodes.